### PR TITLE
fix: fix closing stack using inverted gesture.

### PR DESCRIPTION
**Pull request by AlexeyVlasenko**
Allow you to close Stack that has inverted gesture config using correct closing bool calculation.

### CODE REVIEW

**Comment by AlexeyVlasenko** [comment]
> Thanks for the PR. Is `Math.abs` the correct solution or we should multiply with `-1` instead?
> 
> cc @tienph6m who removed `Math.abs` here #7700

I think both Math.abs and multiply with -1 will produce equal performance, so Math.abs look better there

**Comment by satya164** [comment]
> I think both Math.abs and multiply with -1 will produce equal performance

It's not about performance. The linked PR fixed a bug by removing `Math.abs`, using it for inverted directions will likely re-introduce that bug for inverted directions.

**Comment by AlexeyVlasenko** [comment]
> > I think both Math.abs and multiply with -1 will produce equal performance
> 
> It's not about performance. The linked PR fixed a bug by removing `Math.abs`, using it for inverted directions will likely re-introduce that bug for inverted directions.

No, it wouldn't, because in my fix i use Math.abs only for  INVERTED config.

**Comment by satya164** [comment]
> No, it wouldn't, because in my fix i use Math.abs only for INVERTED config.

So the bug will only happen only for INVERTED config like I said.

**Comment by tienph6m** [comment]
Hi, sorry I wasn't aware of the gestureDirection option when making that PR.
I think using Math.abs in this case would re-introduce the bug like @satya164 said, we should multiple the values with `-1`.

**Comment by AlexeyVlasenko** [comment]
> Hi, sorry I wasn't aware of the gestureDirection option when making that PR.
> I think using Math.abs in this case would re-introduce the bug like @satya164 said, we should multiple `translation + velocity * gestureVelocityImpact` with `-1` when the gesture direction is inverted, probably don't modify `velocity` here because we still need to use the original `velocity` to perform the closing animation.

I can't reproduce bug mentioned in [#7700](https://github.com/react-navigation/react-navigation/pull/7700) using this fix, can you check it, please?

File: packages/stack/src/views/Stack/Card.tsx [hunk 1]
```diff
@@ -271,6 +271,14 @@ export default class Card extends React.Component<Props> {
           velocity = nativeEvent.velocityX;
         }
 
+        if (
+          gestureDirection === 'horizontal-inverted' ||
+          gestureDirection === 'vertical-inverted'
+        ) {
+          translation = Math.abs(translation);
+          velocity = Math.abs(translation);
```
Comment (tienph6m) [inline packages/stack/src/views/Stack/Card.tsx]: I think you meant `velocity = Math.abs(velocity);`?

Comment (AlexeyVlasenko) [reply packages/stack/src/views/Stack/Card.tsx]: Yes, my bad, sorry.

**Comment by tienph6m** [comment]
Mar-19-2020 16-13-49
I was able to reproduce the bug pretty easily, the bug occurs when you swipe back really fast.

I think the directions of `translation` and `velocity` matter and stripping them away with `Math.abs` will cause unexpected behaviors. You can try multiplying `translation + velocity * gestureVelocityImpact` with `-1` when the gesture is inverted and see that it will behave correctly.

File: packages/stack/src/views/Stack/Card.tsx [hunk 2]
```diff
@@ -271,6 +271,14 @@ export default class Card extends React.Component<Props> {
           velocity = nativeEvent.velocityX;
         }
 
+        if (
+          gestureDirection === 'horizontal-inverted' ||
+          gestureDirection === 'vertical-inverted'
+        ) {
+          translation = Math.abs(translation);
```
Comment (satya164) [inline packages/stack/src/views/Stack/Card.tsx]:
```suggestion
          translation *= -1;
```

File: packages/stack/src/views/Stack/Card.tsx [hunk 3]
```diff
@@ -271,6 +271,14 @@ export default class Card extends React.Component<Props> {
...
+          velocity = Math.abs(velocity);
```
Comment (satya164) [inline packages/stack/src/views/Stack/Card.tsx]:
```suggestion
          velocity *= -1;
```

**Pull request merged**